### DISCUSSION TITLE
Add Innerblock Programatically.

### DISCUSSION THREAD

**Discussion by abhi3315** [WordPress/gutenberg#55630] [Q&A]
I developed two custom Gutenberg blocks - `custom/tabs` as the parent block, and `custom/tab` (which uses InnerBlock to add tab content) as the child block. I need to insert the child block using code, and I want to create a button in the Inspector control of `custom/tabs` that allows me to add the child block without affecting the content of any already-added `custom/tab` blocks.

**Comment by stokesman** [comment]
> I need to insert the child block using code

I wonder about this. It’s not clear to me why `custom/tab` can’t have a [`template`](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/nested-blocks-inner-blocks/#template) defined so the child block is automatically inserted along with the `custom/tab`. 

If you actually do need to insert it imperatively, start with [`createBlock`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-blocks/#createblock) and pass its result to [`insertBlock`](https://developer.wordpress.org/block-editor/reference-guides/data/data-core-block-editor/#insertblock). Something like:
```jsx
// …admist your custom/tabs edit function…
const { insertBlock } = useDispatch(blockEditorStore);
const insertTabContent = () => {
	const newBlock = createBlock( 'core/paragraph' , { placeholder: 'this is your tab content…' } );
	insertBlock( newBlock, 0, clientIdOfTabBlock )
}
return (
	// …amidst the inspector controls…
	<Button onClick={ insertTabContent }>Content me</Button>
);
```
